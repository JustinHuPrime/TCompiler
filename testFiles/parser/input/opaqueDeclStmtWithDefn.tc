module foo;

void bar() {
  opaque o;
  struct o {
    int x, y;
  };
  o *ptr = null;
}