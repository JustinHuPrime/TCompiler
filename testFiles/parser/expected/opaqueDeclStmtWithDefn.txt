testFiles/parser/input/opaqueDeclStmtWithDefn.tc (code):
FILE(1, 1, STAB(ENTRY(bar, FUNCTION(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 3, 1, void()))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), FUNDEFN(3, 1, KEYWORDTYPE(3, 1, void), ID(3, 6, bar, REFERENCES(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 3, 1)), STAB(), COMPOUNDSTMT(3, 12, STAB(ENTRY(o, STRUCT(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 4, 3, FIELD(int, x), FIELD(int, y))), ENTRY(ptr, VARIABLE(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 8, 6, o *))), OPAQUEDECL(4, 3, ID(4, 10, o, REFERENCES(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 4, 3))), STRUCTDECL(5, 3, ID(5, 10, o, REFERENCES(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 4, 3)), VARDECL(6, 5, KEYWORDTYPE(6, 5, int), ID(6, 9, x, REFERENCES()), ID(6, 12, y, REFERENCES()))), VARDEFNSTMT(8, 3, MODIFIEDTYPE(8, 3, POINTER, ID(8, 3, o, REFERENCES(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 4, 3))), ID(8, 6, ptr, REFERENCES(testFiles/parser/input/opaqueDeclStmtWithDefn.tc, 8, 6)), LITERAL(8, 12, NULL())))))
